gcloud-java-contrib/gcloud-java-nio/src/main/java/com/google/gcloud/storage/contrib/nio/CloudStorageObjectAttributes.java

```diff
@@ -131,25 +145,27 @@ public FileTime lastAccessTime() {
 
   @Override
   public Object fileKey() {
-    return metadata.getFilename();
+    return name;
```

Comment:
I was trying to optimize for simplicity but, yes, we could provide more than the spec requires. As @aozarov pointed out we can use `id` for the purpose you describe.

Since you're pushing this I actually added a test for filekey. This reveals that we are returning null for directories which arguably disobeys the spec. So I'm updating the code to have a meaningful id for (pseudo) directories (when enabled, of course).